internal/rbd/mirror.go

```diff
@@ -25,6 +25,68 @@ import (
 	librbd "github.com/ceph/go-ceph/rbd"
 )
 
+// FlattenMode is used to indicate the flatten mode for an RBD image.
+type FlattenMode string
+
+const (
+	// FlattenModeNever indicates that the image should never be flattened.
+	FlattenModeNever FlattenMode = "never"
+	// FlattenModeForce indicates that the image with parent should definitely
+	// be flattened.
+	FlattenModeForce FlattenMode = "force"
+)
+
+// HandleParentImage checks the image's parent.
+// if the parent image does not exist and is not in trash, it returns nil.
+// if the flattenMode is FlattenModeForce, it flattens the image itself.
+// if the parent image is in trash, it returns an error.
+// if the parent image exists and is not enabled for mirroring, it returns an error.
+func (rv *rbdVolume) HandleParentImage(
```

Comment:
should we rename this to MirrorImageIfParentExists or something else as we are not handling the ParentImage rather we are flattening of the image only if the parent exists ?

Reply:
HandleParentImage can also be read as handling the cases where parent image may exist and the method to handle such parent depends.
Currently we either error out or flatten the live image.

Reply:
we are acting/handling on the actual image not on the parent, IMO we don't handle anything in parentImage we are having few checks on it, the main idea is to flatten the main/live image. For me, it make sense to have something like `MirrorImageIfParentExists` and `HandleParentImage` looks to be generic and doesn't mention what exactly we are handing here. please feel free to rename to anything else that gives a sense of what we are doing in this function.

Reply:
Renamed it to `HandleParentImageExistence` to imply this function contains logic to handle existence of a parent
image